pkg/virt-controller/watch/migration/migration.go

```diff
@@ -1381,8 +1577,23 @@ func (c *Controller) sync(key string, migration *virtv1.VirtualMachineInstanceMi
 		return nil
 	}
 
-	if migrationFinalizedOnVMI := vmi.Status.MigrationState != nil && vmi.Status.MigrationState.MigrationUID == migration.UID &&
+	if migrationFinalizedOnVMI := vmi.IsMigrationSynchronized(migration) && vmi.Status.MigrationState.MigrationUID == migration.UID &&
 		vmi.Status.MigrationState.EndTimestamp != nil; migrationFinalizedOnVMI {
+		if migration.IsDecentralizedSource() {
+			log.Log.Object(migration).Object(vmi).V(2).Infof("successfuly migrated, cleaning up source virtual machine instance")
+			// Delete the source VMI since we have migration to another cluster/namespace
+			if err := c.clientset.VirtualMachine(vmi.Namespace).Stop(context.Background(), vmi.Name, &virtv1.StopOptions{}); err != nil {
```

Comment:
Why do we have to stop the VM (or delete the VMI)? Shouldn't the VMI successfully terminate on its own? I'm concerned this will mess with the runStrategy logic...

Reply:
Do you know if goes into failed? That seems like a bug to me (albeit potentially fixed by https://github.com/kubevirt/kubevirt/pull/14705)